lib/easybill_rest_client/api_client.rb

```diff
@@ -26,55 +30,52 @@ def request_collection(method, endpoint, params = {})
     end
 
     def request(method, endpoint, params = {})
-      Retryable.retryable(
-        tries: tries,
-        sleep: retry_cool_off_time,
-        on: EasybillRestClient::TooManyRequests
-      ) do
-        response = perform_request(method, endpoint, params)
-        process_response(response)
+      @request_logger = RequestLogger.new(logger: logger, request_id: SecureRandom.hex(3))
```

Comment:
I could run `ApiClient.request(foo,bar)` 2 times in parallel from my app. But there is only one instance variable `@request_logger` which also includes state for each request (the RequestID). So it gets overwritten by the second request while the first request might still be running. Then the first request would use the request logger of the second request to do its logging.